hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/CryptoDeleteHandlerTest.java

```diff
@@ -27,169 +27,157 @@
+        assertEquals(key, context.payerKey());
         basicMetaAssertions(context, 0);
     }
 
     @Test
     void preHandlesCryptoDeleteIfReceiverSigRequiredVanilla() throws PreCheckException {
-        final var keyUsed = (JKey) payerKey;
-
-        given(accounts.get(EntityNumVirtualKey.fromLong(deleteAccountNum))).willReturn(deleteAccount);
-        given(accounts.get(EntityNumVirtualKey.fromLong(transferAccountNum))).willReturn(transferAccount);
-        given(deleteAccount.getAccountKey()).willReturn(keyUsed);
-        given(transferAccount.getAccountKey()).willReturn(keyUsed);
-        given(transferAccount.isReceiverSigRequired()).willReturn(true);
+        given(transferAccount.key()).willReturn(key);
+        given(transferAccount.receiverSigRequired()).willReturn(true);
+        given(deleteAccount.key()).willReturn(accountKey);
 
         final var txn = deleteAccountTransaction(deleteAccountId, transferAccountId);
 
-        final var context = new PreHandleContext(store, txn);
+        final var context = new PreHandleContext(readableStore, txn);
         subject.preHandle(context);
 
         assertEquals(txn, context.body());
         basicMetaAssertions(context, 0);
-        assertEquals(payerKey, context.payerKey());
+        assertEquals(key, context.payerKey());
     }
 
     @Test
     void doesntAddBothKeysAccountsSameAsPayerForCryptoDelete() throws PreCheckException {
-        final var txn = deleteAccountTransaction(payer, payer);
+        final var txn = deleteAccountTransaction(id, id);
 
-        final var context = new PreHandleContext(store, txn);
+        final var context = new PreHandleContext(readableStore, txn);
         subject.preHandle(context);
 
         assertEquals(txn, context.body());
         basicMetaAssertions(context, 0);
-        assertEquals(payerKey, context.payerKey());
+        assertEquals(key, context.payerKey());
         assertIterableEquals(List.of(), context.requiredNonPayerKeys());
     }
 
     @Test
     void doesntAddTransferKeyIfAccountSameAsPayerForCryptoDelete() throws PreCheckException {
-        final var keyUsed = (JKey) payerKey;
+        final var txn = deleteAccountTransaction(deleteAccountId, id);
 
-        given(accounts.get(EntityNumVirtualKey.fromLong(deleteAccountNum))).willReturn(deleteAccount);
-        given(deleteAccount.getAccountKey()).willReturn(keyUsed);
-
-        final var txn = deleteAccountTransaction(deleteAccountId, payer);
+        readableAccounts = emptyReadableAccountStateBuilder()
+                .value(EntityNumVirtualKey.fromLong(accountNum), account)
+                .value(EntityNumVirtualKey.fromLong(deleteAccountNum), deleteAccount)
+                .value(EntityNumVirtualKey.fromLong(transferAccountNum), transferAccount)
+                .build();
+        given(readableStates.<EntityNumVirtualKey, Account>get(ACCOUNTS)).willReturn(readableAccounts);
+        readableStore = new ReadableAccountStore(readableStates);
```

Comment:
Assigning a single base variable inside of multiple tests like this makes it hard to run tests in parallel. Not a huge issue, but we may want to revisit this

Reply:
Thanks for the suggestion @mhess-swl . I will keep it in mind and add different methods to separate them out in next PRs.